openmeter/productcatalog/subscription/service/service.go

```diff
@@ -28,3 +30,21 @@ func New(c Config) plansubscription.PlanSubscriptionService {
 		Config: c,
 	}
 }
+
+func (s *service) respectStartingPhase(p *plan.Plan, startingPhase string) error {
```

Comment:
Maybe:

```suggestion
func (s *service) removePhasesBeforeStartingPhase(p *plan.Plan, startingPhase string) error {
```

For me it's hard to understand what it means to respect the starting phase. If the function name is more verbose, then it becomes self-explanatory.